packages/suite-web/e2e/tests/suite/passphrase-reconnection.test.ts

```diff
@@ -0,0 +1,117 @@
+// @group_passphrase
+// @retry=2
+import { Requests } from '../../support/types';
+
+let requests: Requests;
+
+const abcAddr = 'bc1qpyfvfvm52zx7gek86ajj5pkkne3h385ada8r2y';
+
+describe('Passphrase reconnection', () => {
+    beforeEach(() => {
+        // note that versions before 2.3.1 don't have passphrase caching, this means that returning
+        // back to passphrase that was used before in the session would require to type the passphrase again
+        cy.task('startEmu', { wipe: true });
+        cy.task('setupEmu', {
+            mnemonic: 'mnemonic_all',
+            passphrase_protection: true,
+        });
+        cy.task('startBridge');
+
+        cy.viewport('macbook-13').resetDb();
+        cy.prefixedVisit('/');
+        cy.passThroughInitialRun();
+        cy.discoveryShouldFinish();
+
+        requests = [];
+    });
+
+    it('add 1st hidden wallet (abc) -> succeed -> check 1st address -> disconnect device -> reconnect device -> go back to 1st hidden wallet -> check confirm passphrase appears. ', () => {
+        cy.log('passphrase abc for the first time');
+        // add 1st hidden wallet
+        cy.getTestElement('@menu/switch-device').click();
+        cy.addHiddenWallet('abc');
+
+        cy.interceptDataTrezorIo(requests);
```

Comment:
for example this - we don't care in this test I guess. Maybe this test could be merged with some of the other passphrase related tests.